### PR TITLE
Do not upgrade module on module installation process

### LINKED ISSUE #25570: CI is failing - integration tests - php8 Required parameter $id_shop follows optional parameter $id_linksmenutop
#### Describe the bug

CI is failing on Integration tests with PHP8
```
--- Failed steps:

001 Scenario: Create a simple customer and edit it                         # Features/Scenario/Customer/customer_management.feature:11
      When I edit customer "CUST-1" and I change the following properties: # Features/Scenario/Customer/customer_management.feature:21
        8192: Required parameter $id follows optional parameter $token in /home/runner/work/PrestaShop/PrestaShop/modules/ps_emailsubscription/ps_emailsubscription.php line 316

002 Scenario: Edit category                                        # Features/Scenario/Category/category_management.feature:15
      Given I add new category "category1" with following details: # Features/Scenario/Category/category_management.feature:9
        8192: Required parameter $id_shop follows optional parameter $id_linksmenutop in /home/runner/work/PrestaShop/PrestaShop/modules/ps_mainmenu/ps_menutoplinks.class.php line 29

808 scenarios (806 passed, 2 failed)
13734 steps (13729 passed, 2 failed, 3 skipped)
21m39.28s (343.93Mb)

Script @php -d date.timezone=UTC ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml  --format progress --no-snippets --strict handling the integration-behaviour-tests event returned with error code 1``

Error: Process completed with exit code 1.
```

#### Expected behavior

CI should be green ... 😆 

#### Steps to Reproduce

See Continuous Integration report from any PR

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | We shouldn't upgrade module to latest version when installing a module already installed. THe upgrade process is a separated process
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25570
| How to test?      | CI is green.
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25572)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by kpodemski** [comment]
It was like this for ages. To upgrade the module, you had to upload it again. I'm afraid that this will have some not directly visible issues for some modules, especially those that require some database changes or changes related to the file structure of the module.

I'm not saying it's wrong, but module developers need to plan their next upgrades accordingly. This has to be properly communicated.

**Comment by kpodemski** [comment]
cc @Matt75 wdyt?

**Comment by PierreRambaud** [comment]
> It was like this for ages. To upgrade the module, you had to upload it again. I'm afraid that this will have some not directly visible issues for some modules, especially those that require some database changes or changes related to the file structure of the module.
> 
> I'm not saying it's wrong, but module developers need to plan their next upgrades accordingly. This has to be properly communicated.

We shouldn't upgrade something we are trying to install, or at least only upgrade if it came from a zip file and the version is newer than the previous.

**Comment by matks** [comment]
> It was like this for ages. To upgrade the module, you had to upload it again. I'm afraid that this will have some not directly visible issues for some modules, especially those that require some database changes or changes related to the file structure of the module.
> 
> I'm not saying it's wrong, but module developers need to plan their next upgrades accordingly. This has to be properly communicated.

The behavior today is:

When PrestaShop attempts to install module A version Y
If module A version X is already installed
Then perform the upgrade OR downgrade from version X to version Y

It can be useful, but it's also very unreliable 😄 .

I agree with @PierreRambaud that a function `install()` should not perform an `upgrade()` inside its logic.

We can imagine a better way to provide the above behavior explicitely. For example
- provide a `install()` function that throws an Exception if module is already installed
- add a `installOrUpgrade()` function
- add a `updateToNewVersion($allowsDowngrade)` function that you must expliticitely allow to downgrade

Also, when shop admin uploads a new ZIP of module A (with version Y) to its folder, we could add a new step.

If module is not installed nor present on disk: show a information message "Module A is not installed nor present on disk, please confirm you want to install it"
If module is not installed but present on disk: show a information message "Module A is present on disk, but not installed please confirm you want to replace it"
If module is installed with version X and X < Y: show a information message "Module A is already installed please confirm you want to upgrade it from version X to version Y"
If module is installed with version X and X > Y: show a information message "Module A is already installed please confirm you want to downgrade it from version X to version Y"

**Comment by kpodemski** [comment]
Again. I'm not saying it's wrong to remove it from `install(),` but:

1. we need to communicate it properly as it's a quite big change
2. we need to have an alternative way to run an upgrade for modules uploaded from the back-office, as uploading them is the only easy way for merchants to update them - the only way for many, because an alternative is to upload the module via FTP 🙄 (of course I'm not writing about modules that have some built-in upgrade mechanism, It's not allowed on the Addons Marketplace for example)

I like this idea:

> If module A version X is already installed
> Then perform the upgrade OR downgrade from version X to version Y
> It can be useful, but it's also very unreliable 😄 .

But, let's not think about downgrading anything because this could have introduced some serious issues. Let's think about a solution to upgrade the module to a newer version 👍 

Remember that uploading the module in PrestaShop replaces all the files. This is why if we don't run an upgrade, the store would be broken because, for example, one of the tables in the database has an outdated structure.

This is why if there is no alternative now, I'm highly against this change.

**Comment by PierreRambaud** [comment]
I updated my code to make it simpler:
If the module is installed and if the source is a zip file, try to upgrade the module, otherwise do nothing :)

**Review by Progi1984 — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
               restore-keys: ${{ runner.os }}-composer-
 
           - name: Composer Install
-            run: composer install --ansi --prefer-dist --no-interaction --no-progress --quiet
+            run: composer install --ansi --prefer-dist --no-interaction --no-progress
```
Comment (sowbiba) [inline .github/workflows/integration.yml:57]: This is temporary right?

File: composer.json [hunk 2]
```diff
@@ -195,7 +195,7 @@
         "git-hook-install": "@php .github/contrib/install.php",
         "git-hook-uninstall": "@php .github/contrib/uninstall.php",
         "integration-behaviour-tests": [
-            "@php -d date.timezone=UTC ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml  --format progress --no-snippets --strict"
+            "@php -d date.timezone=UTC ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml  --format progress --no-snippets --strict --stop-on-failure"
```
Comment (sowbiba) [inline composer.json]: This is temporary right ?

**Review by sowbiba — COMMENTED** [review]
LGTM if the answers for my 2 questions are affirmative

**Review by matthieu-rolland — DISMISSED** [review]

File: composer.json (hunk 2 above)
Comment (PierreRambaud) [reply composer.json]: Yes it is, my bad, I revert this change

File: .github/workflows/integration.yml (hunk 1 above)
Comment (PierreRambaud) [reply .github/workflows/integration.yml:57]: No, we want to keep information when running under CI context

**Review by matks — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

File: src/Core/Addon/Module/ModuleManager.php [hunk 3]
```diff
@@ -288,7 +288,12 @@ public function install($source)
         }
 
         if ($this->moduleProvider->isInstalled($name)) {
-            return $this->upgrade($name, 'latest', $source);
+            if ($source !== null) {
+                return $this->upgrade($name, 'latest', $source);
+            }
```
Comment (eternoendless) [inline src/Core/Addon/Module/ModuleManager.php:293]: IMO install should never upgrade on its own.

Comment (PierreRambaud) [reply src/Core/Addon/Module/ModuleManager.php:293]: Until you're trying to install a new zip, that's what @kpodemski saying above :thinking:

**Review by kpodemski — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @PierreRambaud

**Pull request merged**